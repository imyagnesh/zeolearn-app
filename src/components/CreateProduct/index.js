import React, { Component } from 'react';
import PropTypes from 'prop-types';
import wrapper from '../../HOC/wrapper';

const styles = {
  wrapper: {
    margin: '5px',
  },
};

class CreateProduct extends Component {
  static propTypes = {
    product: PropTypes.object.isRequired,
    onSubmitForm: PropTypes.func.isRequired,
    onChangeText: PropTypes.func.isRequired,
  };

  render() {
    const { product, onSubmitForm, onChangeText } = this.props;
    return (
      <form
        style={{
          display: 'flex',
          flex: 1,
          flexDirection: 'column',
        }}
        onSubmit={e => onSubmitForm(e)}
      >
        <h1>{this.props.add(1, 3)}</h1>
        <input type="hidden" value={product.id} name="id" />
        <input
          ref={(ref) => {
            this.productName = ref;
          }}
          style={styles.wrapper}
          type="text"
          value={product.productName}
          name="productName"
          onChange={e => onChangeText(e)}
        />
        <input
          style={styles.wrapper}
          type="text"
          value={product.productType}
          name="productType"
          onChange={e => onChangeText(e)}
        />
        <input
          style={styles.wrapper}
          type="number"
          value={product.price}
          name="price"
          onChange={e => onChangeText(e)}
        />
        <input style={styles.wrapper} type="submit" value="Submit" />
      </form>
    );
  }
}

export default wrapper(CreateProduct);
