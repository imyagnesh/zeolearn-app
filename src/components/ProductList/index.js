import React, { Component } from 'react';
import PropTypes from 'prop-types';
import wrapper from '../../HOC/wrapper';

class ProductList extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { data, updateProduct, deleteProduct } = this.props;
    return (
      <div>
        <h1>{this.props.title}</h1>
        <table
          style={{
            width: '100%',
          }}
        >
          <thead>
            <tr>
              <th>ID</th>
              <th>Product Name</th>
              <th>Product Type</th>
              <th>Price</th>
              <th>Actions</th>
            </tr>
          </thead>
          <tbody>
            {data.map(item => (
              <tr key={item.id}>
                <td>{item.id}</td>
                <td>{item.productName}</td>
                <td>{item.productType}</td>
                <td>{item.price}</td>
                <td>
                  <input type="button" value="edit" onClick={() => updateProduct(item)} />
                  <input type="button" value="delete" onClick={() => deleteProduct(item)} />
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

ProductList.propTypes = {
  data: PropTypes.array.isRequired,
  updateProduct: PropTypes.func.isRequired,
  deleteProduct: PropTypes.func.isRequired,
};

export default wrapper(ProductList);
