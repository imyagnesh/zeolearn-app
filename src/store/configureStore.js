import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import rootReducer from "../reducers";

let middleware = [thunk];

middleware = [...middleware];

export default function configureStore(initialState) {
  const store = createStore(
    rootReducer,
    initialState,
    compose(applyMiddleware(...middleware))
  );
  return store;
}
