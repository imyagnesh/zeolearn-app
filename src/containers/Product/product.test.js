import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import CreateProduct from "../../components/CreateProduct";
import ProductList from "../../components/ProductList";

import configureStore from "../../store/configureStore";

import { Product, initialProduct } from "./index";

const store = configureStore();

configure({ adapter: new Adapter() });

function setup() {
  const props = {
    products: {
      loading: false,
      data: [],
      error: false
    },
    actions: {
      getProducts: jest.fn(),
      saveProducts: jest.fn()
    }
  };
  const enzymeWrapper = shallow(<Product store={store} {...props} />);
  return {
    props,
    enzymeWrapper
  };
}

describe("Product", () => {
  it("renders correctly", () => {
    const { props } = setup();
    const tree = renderer.create(<Product {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render self and subcomponents", () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find(CreateProduct).exists()).toBe(true);
    expect(enzymeWrapper.find(ProductList).exists()).toBe(true);
    const createProduct = enzymeWrapper.find(CreateProduct).props();
    expect(createProduct.product).toEqual(initialProduct);
  });

  it("should search product", () => {
    const { enzymeWrapper, props } = setup();
    const form = enzymeWrapper.find("form").props();
    form.onSubmit({ preventDefault: jest.fn() });
    expect(props.actions.saveProducts.mock.calls.length).toBe(1);
    form.onSubmit({ preventDefault: jest.fn() });
    expect(props.actions.saveProducts.mock.calls.length).toBe(2);
  });
});
