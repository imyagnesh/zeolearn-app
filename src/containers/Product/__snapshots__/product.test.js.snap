// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
<div>
  <h1>
    Product
  </h1>
  <div
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <h2>
      Hello John
    </h2>
    <form
      onSubmit={[Function]}
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <h1>
        4
      </h1>
      <input
        name="id"
        type="hidden"
        value=""
      />
      <input
        name="productName"
        onChange={[Function]}
        style={
          Object {
            "margin": "5px",
          }
        }
        type="text"
        value=""
      />
      <input
        name="productType"
        onChange={[Function]}
        style={
          Object {
            "margin": "5px",
          }
        }
        type="text"
        value=""
      />
      <input
        name="price"
        onChange={[Function]}
        style={
          Object {
            "margin": "5px",
          }
        }
        type="number"
        value={0}
      />
      <input
        style={
          Object {
            "margin": "5px",
          }
        }
        type="submit"
        value="Submit"
      />
    </form>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="search"
      onChange={[Function]}
      type="text"
    />
    <input
      type="submit"
      value="Search"
    />
  </form>
  <div
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <h2>
      Hello John
    </h2>
    <div>
      <h1>
        Products
      </h1>
      <table
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              Product Name
            </th>
            <th>
              Product Type
            </th>
            <th>
              Price
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
